process/guardian/guardedAccount.go

```diff
@@ -101,6 +112,43 @@ func (agc *guardedAccount) setAccountGuardian(uah state.UserAccountHandler, guar
 	return agc.saveAccountGuardians(accHandler, newGuardians)
 }
 
+func (agc *guardedAccount) instantSetGuardian(
+	uah state.UserAccountHandler,
+	guardianAddress []byte,
+	txGuardianAddress []byte,
+) error {
+	accountGuardians, err := agc.getConfiguredGuardians(uah)
+	if err != nil {
+		return err
+	}
+
+	activeGuardian, err := agc.getActiveGuardian(accountGuardians)
+	if err != nil {
+		return err
+	}
+
+	if !bytes.Equal(activeGuardian.Address, txGuardianAddress) {
+		return process.ErrTransactionAndAccountGuardianMismatch
+	}
+
+	// immediately set the new guardian
+	guardian := &guardians.Guardian{
+		Address:         guardianAddress,
+		ActivationEpoch: agc.currentEpoch,
+	}
+
+	accountGuardians.Slice = []*guardians.Guardian{guardian}
+	accHandler, ok := uah.(vmcommon.UserAccountHandler)
+	if !ok {
+		return process.ErrWrongTypeAssertion
+	}
+
+	return agc.saveAccountGuardians(accHandler, accountGuardians)
+}
+
+// TODO: add constraints on not co-signed txs on interceptor, for setGuardian
```

Comment:
task for TODO?

Also, question: is there a check as to not send value on a setGuardian transaction call? (in builtin function implementation)

Reply:
Created a task: [EN-12365]

Good point, will add the check on the builtin function in vmcommon, currently there is no check for the value.